drivers/sensor/tmp112/tmp112.c

```diff
@@ -193,19 +201,40 @@ static const struct sensor_driver_api tmp112_driver_api = {
 
 int tmp112_init(struct device *dev)
 {
+	const struct tmp112_config *config = dev->config->config_info;
 	struct tmp112_data *drv_data = dev->driver_data;
 
-	drv_data->i2c = device_get_binding(DT_INST_0_TI_TMP112_BUS_NAME);
+	drv_data->i2c = device_get_binding(config->i2c_master_dev_name);
 	if (drv_data->i2c == NULL) {
 		LOG_DBG("Failed to get pointer to %s device!",
-			    DT_INST_0_TI_TMP112_BUS_NAME);
+			log_strdup(config->i2c_master_dev_name));
 		return -EINVAL;
 	}
 
 	return 0;
 }
 
-static struct tmp112_data tmp112_driver;
+#define TMP112_DEVICE(n)							     \
+	static const struct tmp112_config tmp112_##n##_config = {		     \
+		.i2c_master_dev_name = DT_INST_##n##_TI_TMP112_BUS_NAME,	     \
+		.i2c_slave_addr = DT_INST_##n##_TI_TMP112_BASE_ADDRESS,		     \
+	};									     \
+	static struct tmp112_data tmp112_##n##_driver;				     \
+	DEVICE_AND_API_INIT(tmp112_##n, DT_INST_##n##_TI_TMP112_LABEL, tmp112_init,  \
+			    &tmp112_##n##_driver, &tmp112_##n##_config, POST_KERNEL, \
+			    CONFIG_SENSOR_INIT_PRIORITY, &tmp112_driver_api)
+
+
+TMP112_DEVICE(0);
```

Comment:
For consistency, please wrap in ifdefs like the other two instances

Reply:
It was my first option, but there was a fail from Shippable (warning from gcc) saying that tmp112_driver_api struct was not used, thus the PR was not accepted. Declaring at least one TMP112 removes this warning

Reply:
Which test or sample failed? It would be better to enable an instance of the driver in that test/sample.

Reply:
tests/drivers/build_all/sensors_i_z.conf is failing because no tmp112 sensor is available on some boards...so no dts , so no DT_INST_0_TI_TMP112 ...which make the test failing. And declaring CONFIG_TMP112=n is certainly not the best solution. As soon as a driver is included, at least one instance must be declared...

Reply:
Looks like this might benefit from moving to DT_INST_FOREACH_STATUS_OKAY.

Reply:
OK, I will use new 2.3 macros. Let me take time to rebase first! ;)